dev/helpers/mssql/run_migrations.sh

```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 
-# There seems to be [a bug with docker-compose](https://github.com/docker/compose/issues/4076#issuecomment-324932294) 
+# There seems to be [a bug with docker-compose](https://github.com/docker/compose/issues/4076#issuecomment-324932294)
 # where it takes ~40ms to connect to the terminal output of the container, so stuff logged to the terminal in this time is lost.
 # The best workaround seems to be adding tiny delay like so:
 sleep 0.1;
 
 MIGRATE_DIRECTORY="/mnt/migrator/DbScripts"
+HELPERS_DIRECTORY="/mnt/helpers"
 SERVER='mssql'
 DATABASE="vault_dev"
 USER="SA"
```

Comment:
Can avoid file creation by just using shell variable I think:
```

# Navigate to the migrations directory and create a JSON array of all SQL files
jsonArray=$(cd $MIGRATE_DIRECTORY && ls -1 *.sql | sed 's/^/"/' | sed 's/$/"/' | paste -sd, -)

# Format the JSON array
formattedJsonArray="[$jsonArray]"

# Escape single quotes in JSON content
escapedJsonContent=$(echo $formattedJsonArray | sed "s/'/''/g")

# Prepare the SQL query
query="EXEC ReadRequiredMigrations '$escapedJsonContent'"

# Execute the query using sqlcmd
result=$(/opt/mssql-tools/bin/sqlcmd -S $SERVER -d migrations_$DATABASE -U $USER -P $PASSWD -I -Q "$query" -W -h-1)
echo $result

```

Reply:
This is similar to my initial approach, but I get this error (using your code snippet above):

```bashSqlcmd: '/mnt/helpers/read_required_migrations.sql': Invalid filename.
Msg 105, Level 15, State 1, Server e82f038a69b6, Line 1 Unclosed quotation mark after the character string '["2017-08-19_00_InitialSetup.sql","2017-08-22_00_LicenseCheckScripts.sql","2017-08-30_00_CollectionWriteOnly.sql","2017-09-06_00_CipherDetails.sql","2017-09-08_00_OrgUserCounts.sql","2017-10-25_00_OrgUserUpdates.sql","2017-11-06_00_FamilyPlanAdjustments.sql","2017-11-13_00_IndexTuning.sql","2017-11-24_00_UpdateProcs.sql","2017-12-12_00_Events.sql","2018-02-28_00_LoginUris.sql","2018-03-12_00_FixLoginUris.sql","2018-03-21_00_AdminPortal.sql","2018-04-02_00_Org2fa.sql","2018-04-24_00_CipherQueryTuning.sql","2018-06-11_00_WebVaultUpdates.sql","2018-07-28_00_DbTuning.sql","2018-08-14_00_UserKdf.sql","2018-08-28_00_PremiumOrgAbilities.sql","2018-09-25_00_OrgPurge.sql","2018-10-17_00_ManagerRole.sql","2018-12-19_00_OrgUserTwoFactorEnabled.sql","2019-01-31_00_Transactions.sql","2019-03-01_00_OrgApi.sql","2019-05-01_00_CipherOrgDetailsFix.sql","2019-05-15_00_GroupNameFixes.sql","2019-05-28_00_CollectionCipherImpr
```

There is not actually any unclosed quotation mark, after some googling my best guess was that it's hitting the max command length allowed by the environment and therefore truncating the json string. Hence I switched to importing from a file which can be an arbitrary size/length.

I'm open to suggestions if you know how to solve this error though.

Reply:
This is caused by a broken version of sqlcmd on the tools container that forces textsize session variable of 4096 in the sqlcmd version 13 that it uses. Why not issue the sql commands from powershell like the ef commands but using;
```
 # Load the assembly
Add-Type -AssemblyName "System.Data.SqlClient"
```